code/espurna/config/types.h

```diff
@@ -22,37 +22,48 @@
 // BUTTONS
 //------------------------------------------------------------------------------
 
-#define BUTTON_EVENT_NONE           0
-#define BUTTON_EVENT_PRESSED        1
-#define BUTTON_EVENT_RELEASED       2
-#define BUTTON_EVENT_CLICK          2
-#define BUTTON_EVENT_DBLCLICK       3
-#define BUTTON_EVENT_LNGCLICK       4
-#define BUTTON_EVENT_LNGLNGCLICK    5
-#define BUTTON_EVENT_TRIPLECLICK    6
-
-#define BUTTON_MODE_NONE            0
-#define BUTTON_MODE_TOGGLE          1
-#define BUTTON_MODE_ON              2
-#define BUTTON_MODE_OFF             3
-#define BUTTON_MODE_AP              4
-#define BUTTON_MODE_RESET           5
-#define BUTTON_MODE_PULSE           6
-#define BUTTON_MODE_FACTORY         7
-#define BUTTON_MODE_WPS             8
-#define BUTTON_MODE_SMART_CONFIG    9
-#define BUTTON_MODE_DIM_UP          10
-#define BUTTON_MODE_DIM_DOWN        11
-#define BUTTON_MODE_DISPLAY_ON      12
-
-
-// Needed for ESP8285 boards under Windows using PlatformIO (?)
-#ifndef BUTTON_PUSHBUTTON
-#define BUTTON_PUSHBUTTON           0
-#define BUTTON_SWITCH               1
-#define BUTTON_DEFAULT_HIGH         2
-#define BUTTON_SET_PULLUP           4
-#endif
+// button actions, limited to 4-bit number (0b1111 / 0xf / 15)
+#define BUTTON_ACTION_NONE            0u
+#define BUTTON_ACTION_TOGGLE          1u
+#define BUTTON_ACTION_ON              2u
+#define BUTTON_ACTION_OFF             3u
+#define BUTTON_ACTION_AP              4u
+#define BUTTON_ACTION_RESET           5u
+#define BUTTON_ACTION_PULSE           6u
+#define BUTTON_ACTION_FACTORY         7u
+#define BUTTON_ACTION_WPS             8u
+#define BUTTON_ACTION_SMART_CONFIG    9u
+#define BUTTON_ACTION_DIM_UP          10u
+#define BUTTON_ACTION_DIM_DOWN        11u
+#define BUTTON_ACTION_DISPLAY_ON      12u
+#define BUTTON_ACTION_MAX             15u
+
+// Deprecated: legacy mapping, changed to action from above
+#define BUTTON_MODE_NONE              BUTTON_ACTION_NONE
+#define BUTTON_MODE_TOGGLE            BUTTON_ACTION_TOGGLE
+#define BUTTON_MODE_ON                BUTTON_ACTION_ON
+#define BUTTON_MODE_OFF               BUTTON_ACTION_OFF
+#define BUTTON_MODE_AP                BUTTON_ACTION_AP
+#define BUTTON_MODE_RESET             BUTTON_ACTION_RESET
+#define BUTTON_MODE_PULSE             BUTTON_ACTION_PULSE
+#define BUTTON_MODE_FACTORY           BUTTON_ACTION_FACTORY
+#define BUTTON_MODE_WPS               BUTTON_ACTION_WPS
+#define BUTTON_MODE_SMART_CONFIG      BUTTON_ACTION_SMART_CONFIG
+#define BUTTON_MODE_DIM_UP            BUTTON_ACTION_DIM_UP
+#define BUTTON_MODE_DIM_DOWN          BUTTON_ACTION_DIM_DOWN
+#define BUTTON_MODE_DISPLAY_ON        BUTTON_ACTION_DISPLAY_ON
```

Comment:
Deprecation path for these is a bit tricky. `-DBUTTON1_CLICK=BUTTON_MODE_DIM_UP` would show `error: ‘BUTTON_MODE_DIM_UP’ was not declared in this scope`, but that's it.

`#pragma GCC poison TOKEN` can restrict token usage in code, assuming poison is set even before custom header is included. One issue with that it won't work with -D... flags, `warning: poisoning existing macro "..."` is the best it could do.